modules/layers/src/solid-polygon-layer/solid-polygon-layer-vertex-main.glsl.js

```diff
@@ -52,6 +51,13 @@ vec3 project_offset_normal(vec3 vector) {
 }
 
 void calculatePosition(PolygonProps props) {
+#ifdef IS_SIDE_VERTEX
```

Comment:
Recommend adding comment here to ensure maintainers understand what we are doing.

I am a little worried about degenerate triangles that they can cause some rendering issues on some GPUs. But if all verts are zero it should probably be OK.

How does vertexValid work? Is it always false for all three verts in a triangle - so that we don't just set one vert to 0?

Reply:
In this case (if I'm not missing something) we do instanced rendering, create 4 vertices from prev & next vertex, and all 4 vertices share same vertexValid value.

Reply:
I think it is worth adding this to comments in the shader